drivers/spi/spi_nrfx_spim.c

```diff
@@ -685,6 +769,14 @@ static int spi_nrfx_init(const struct device *dev)
 			(0))),								 \
 		(0))
 
+#define SPIM_INIT_LEVEL(idx) \
+	COND_CODE_1(SPIM_REQUESTS_CLOCK(idx), (POST_KERNEL), (PRE_KERNEL_1))
+
+#define SPIM_INIT_PRIO(idx) \
+	COND_CODE_1(SPIM_REQUESTS_CLOCK(idx), \
+		(UTIL_INC(CONFIG_CLOCK_CONTROL_NRF2_GLOBAL_HSFLL_INIT_PRIORITY)), \
+		(CONFIG_SPI_INIT_PRIORITY))
```

Comment:
This is hiding logic that should be covered by Kconfig - it may be confusing to users.
Perhaps it could be done in Kconfig if init_level and init_priority are defined per-instance?

Reply:
I don't know if Kconfig would be the right place since we are looking at which clock the SPI instance is using, which I guess we could do from the Kconfig as well using kconfig macros, but I really don't know if its worth it if this solution is sufficient, at least for now :)

Reply:
At least we should document in `CONFIG_SPI_INIT_PRIORITY` that it is not applied in certain circumstances.

Reply:
That is fair, adding a comment to the macros `SPIM_INIT_LEVEL()` and `SPIM_INIT_PRIO()` explaining why they are there maybe

Reply:
Added a comment explaining the situation. And corrected this code a little (the `PRE_KERNEL_1` level was obviously wrong here).